src/nuclide_header.F90

```diff
@@ -0,0 +1,820 @@
+          xs = this % total(g)
+        case('absorption')
+          xs = this % absorption(g)
+        case('fission')
+          xs = this % fission(g)
+        case('k_fission')
+          if (allocated(this % k_fission)) then
+            xs = this % k_fission(g)
+          end if
+        case('chi')
+          xs = this % chi(g)
+        case('scatter')
+          xs = this % total(g) - this % absorption(g)
+        end select
+      end if
+    end function nuclide_iso_get_xs
+
+    function nuclide_angle_get_xs(this, g, xstype, gout, uvw, mu, i_azi, i_pol) &
+         result(xs)
+      class(Nuclide_Angle), intent(in) :: this
+      integer, intent(in)              :: g      ! Incoming Energy group
+      character(*), intent(in)         :: xstype ! Cross Section Type
+      integer, optional, intent(in)    :: gout   ! Outgoing Group
+      real(8), optional, intent(in)    :: mu     ! Change in angle
+      real(8), optional, intent(in)    :: uvw(3) ! Requested Angle
+      integer, optional, intent(in)    :: i_azi  ! Azimuthal Index
+      integer, optional, intent(in)    :: i_pol  ! Polar Index
+      real(8)                          :: xs     ! Resultant xs
+
+      integer :: i_azi_, i_pol_
+
+      xs = ZERO
+
+      if ((xstype == 'nu_fission' .or. xstype == 'fission' .or. xstype =='chi' &
+           .or. xstype =='k_fission') .and. (.not. this % fissionable)) then
+        return
+      end if
+
+      if (present(i_azi) .and. present(i_pol)) then
+        i_azi_ = i_azi
+        i_pol_ = i_pol
+      else
+        call find_angle(this % polar, this % azimuthal, uvw, i_azi_, i_pol_)
+      end if
+
+      if (present(gout)) then
+        select case(xstype)
+        case('mult')
+          xs = this % mult(gout,g,i_azi_,i_pol_)
+        case('nu_fission')
+          xs = this % nu_fission(gout,g,i_azi_,i_pol_)
+        case('chi')
+          xs = this % chi(gout,i_azi_,i_pol_)
+        case('f_mu', 'f_mu/mult')
+          xs = this % calc_f(g, gout, mu, I_AZI=i_azi_, I_POL=i_pol_)
+          if (xstype == 'f_mu/mult') then
+            xs = xs / this % mult(gout,g,i_azi_,i_pol_)
+          end if
+        end select
+      else
+        select case(xstype)
+        case('total')
+          xs = this % total(g,i_azi_,i_pol_)
+        case('absorption')
+          xs = this % absorption(g,i_azi_,i_pol_)
+        case('fission')
+          xs = this % fission(g,i_azi_,i_pol_)
+        case('k_fission')
+          if (allocated(this % k_fission)) then
+            xs = this % k_fission(g,i_azi_,i_pol_)
+          end if
+        case('chi')
+          xs = this % chi(g,i_azi_,i_pol_)
+        case('scatter')
+          xs = this % total(g,i_azi_,i_pol_) - this % absorption(g,i_azi_,i_pol_)
+        end select
+      end if
+
+    end function nuclide_angle_get_xs
+
+!===============================================================================
+! NUCLIDE_*_CALC_F Finds the value of f(mu), the scattering probability, given mu
```

Comment:
corrected comment; it is the scattering angular distribution, so i call it the scattering angle probability